content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-13.md

```diff
@@ -0,0 +1,233 @@
+- Check the prerequisites for [Upgrading/downgrading kubeadm clusters between v1.12 to v1.13](/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-13/).
+
+{{< note >}}
+All commands on any control plane or etcd node should be run as root.
+{{< /note >}}
+
+{{% /capture %}}
+
+{{% capture steps %}}
+
+## Prepare for both methods
+
+Upgrade `kubeadm` to the version that matches the version of Kubernetes that you are upgrading to:
+
+```shell
+apt-mark unhold kubeadm && \
+apt-get update && apt-get install -y kubeadm && \
+apt-mark hold kubeadm
+```
+
+Check prerequisites and determine the upgrade versions:
+
+```shell
+kubeadm upgrade plan
+```
+
+You should see something like the following:
+
+```
+Upgrade to the latest version in the v1.13 series:
+
+COMPONENT            CURRENT   AVAILABLE
+API Server           v1.12.2   v1.13.0
+Controller Manager   v1.12.2   v1.13.0
+Scheduler            v1.12.2   v1.13.0
+Kube Proxy           v1.12.2   v1.13.0
+CoreDNS              1.2.2     1.2.6
+```
+
+## Stacked control plane nodes
+
+### Upgrade the first control plane node
+
+Modify `configmap/kubeadm-config` for this control plane node:
+
+```shell
+kubectl get configmap -n kube-system kubeadm-config -o yaml > kubeadm-config-cm.yaml
+```
+
+Open the file in an editor and replace the following values:
+
+- `api.advertiseAddress`
+
+    This should be set to the local node's IP address.
+
+- `etcd.local.extraArgs.advertise-client-urls`
+
+    This should be updated to the local node's IP address.
+
+- `etcd.local.extraArgs.initial-advertise-peer-urls`
+
+    This should be updated to the local node's IP address.
+
+- `etcd.local.extraArgs.listen-client-urls`
+
+    This should be updated to the local node's IP address.
+
+- `etcd.local.extraArgs.listen-peer-urls`
+
+    This should be updated to the local node's IP address.
+
+- `etcd.local.extraArgs.initial-cluster`
+
+    This should be updated to include the hostname and IP address pairs for each control plane node in the cluster. For example:
+
+        "ip-172-31-92-42=https://172.31.92.42:2380,ip-172-31-89-186=https://172.31.89.186:2380,ip-172-31-90-42=https://172.31.90.42:2380"
+
+You must also pass an additional argument (`initial-cluster-state: existing`) to etcd.local.extraArgs.
+
+```shell
+kubectl apply -f kubeadm-config-cm.yaml --force
+```
+
+Start the upgrade:
+
+```shell
+kubeadm upgrade apply v<YOUR-CHOSEN-VERSION-HERE>
+```
+
+You should see something like the following:
+
+    [upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.12.0". Enjoy!
+
+The `kubeadm-config` ConfigMap is now updated from `v1alpha2` version to `v1alpha3`.
+
+### Upgrading additional control plane nodes
+
+Each additional control plane node requires modifications that are different from the first control plane node. Run:
```

Comment:
@rdodev 
In v1.12 additional control-plane nodes are supported only if using external etcd, so I don't think changing etcd.local makes sense (and it might eventually lead to problems, but I'm not sure about this)